src/adapter/src/coord/dataflows.rs

```diff
@@ -97,11 +97,36 @@ impl Coordinator {
 
     /// Finalizes a list of dataflows and then broadcasts it to all workers.
     async fn ship_dataflows(&mut self, dataflows: Vec<DataflowDesc>, instance: ComputeInstanceId) {
+        self.ship_dataflows_fallible(dataflows, instance)
+            .await
+            .expect("failed to ship dataflows");
+    }
+
+    /// Finalizes a dataflow and then broadcasts it to all workers.
+    /// Utility method for the more general [`Self::ship_dataflows_fallible`]
+    ///
+    /// Returns an error on failure. DO NOT call this for DDL.
+    pub(crate) async fn ship_dataflow_fallible(
+        &mut self,
+        dataflow: DataflowDesc,
+        instance: ComputeInstanceId,
+    ) -> Result<(), AdapterError> {
+        self.ship_dataflows_fallible(vec![dataflow], instance).await
+    }
+
+    /// Finalizes a list of dataflows and then broadcasts it to all workers.
+    ///
+    /// Returns an error on failure. DO NOT call this for DDL.
```

Comment:
Should this also error instead of panicking if `create_dataflows` fails?

Reply:
I'm not entirely sure tbh. Giving it a quick skim it seems like some of the errors returned indicate some really bad inconsistent state where there's not much to do but panic, while others indicate a programmer error that doesn't need to cause a panic.

I'm planning on opening a followup issue to audit all adapter panics and see which can be turned into user errors.